packages/nova-react/src/eventing/nova-eventing-provider.tsx

```diff
@@ -241,3 +267,25 @@ const generateEventing =
         : Promise.resolve();
     },
   });
+
+const generateEventing =
+  (
+    eventingRef: React.MutableRefObject<NovaEventing>,
+    interceptorRef?: React.MutableRefObject<
+      (event: EventWrapper) => Promise<EventWrapper | undefined>
+    >,
+  ) =>
+  (): NovaEventing => ({
+    bubble: async (eventWrapper: EventWrapper) => {
```

Comment:
The code 
```
if (!interceptorRef) {
        return eventingRef.current.bubble(eventWrapper);
      }

      let eventToBubble: EventWrapper | undefined = eventWrapper;
      eventToBubble = await interceptorRef.current(eventWrapper);

      return eventToBubble
        ? eventingRef.current.bubble(eventToBubble)
        : Promise.resolve();
```

repeats in 3 places but it could be shared by extracting a function `getBubble(eventingRef, interceptorRef)`. Than instead of create a confusing pair of two functions with similar name, we can just use that function directly in interceptor inlining the creation of lower level eventings

Reply:
good point, updated it